common/types/errors.go

```diff
@@ -89,32 +96,59 @@ func (err RemoteSyncMatchedError) Error() string {
 }
 
 func (err RetryTaskV2Error) Error() string {
-	return fmt.Sprintf("RetryTaskV2Error{Message: %v, DomainId: %v, WorkflowId: %v, RunId: %v, StartEventId: %v, StartEventVersion: %v, EndEventId: %v, EndEventVersion: %v}",
-		err.Message,
-		err.DomainID,
-		err.WorkflowID,
-		err.RunID,
-		err.StartEventID,
-		err.StartEventVersion,
-		err.EndEventID,
-		err.EndEventVersion,
-	)
+	fields := []string{}
+	fields = append(fields, fmt.Sprintf("Message: %v", err.Message))
+	if err.DomainID != nil {
+		fields = append(fields, fmt.Sprintf("DomainID: %v", err.DomainID))
+	}
+	if err.WorkflowID != nil {
+		fields = append(fields, fmt.Sprintf("WorkflowID: %v", err.WorkflowID))
+	}
+	if err.RunID != nil {
+		fields = append(fields, fmt.Sprintf("RunID: %v", err.RunID))
+	}
+	if err.StartEventID != nil {
+		fields = append(fields, fmt.Sprintf("StartEventID: %v", err.StartEventID))
+	}
+	if err.StartEventVersion != nil {
+		fields = append(fields, fmt.Sprintf("StartEventVersion: %v", err.StartEventVersion))
+	}
+	if err.EndEventID != nil {
+		fields = append(fields, fmt.Sprintf("EndEventID: %v", err.EndEventID))
+	}
+	if err.EndEventVersion != nil {
+		fields = append(fields, fmt.Sprintf("EndEventVersion: %v", err.EndEventVersion))
+	}
+	return fmt.Sprintf("RetryTaskV2Error{%s}", strings.Join(fields, ", "))
 }
 
 func (err ServiceBusyError) Error() string {
 	return fmt.Sprintf("ServiceBusyError{Message: %v}", err.Message)
 }
 
 func (err WorkflowExecutionAlreadyStartedError) Error() string {
-	return fmt.Sprintf("WorkflowExecutionAlreadyStartedError{Message: %v, StartRequestId: %v, RunId: %v}",
-		err.Message,
-		err.StartRequestID,
-		err.RunID,
-	)
+	fields := []string{}
```

Comment:
What about using a string buffer instead of an array - I think it will be more efficient.

Reply:
Updated.